src/test/DelegationFaucet.t.sol

```diff
@@ -0,0 +1,219 @@
+
+import "src/test/EigenLayerTestHelper.t.sol";
+
+contract DelegationFaucetTests is EigenLayerTestHelper {
+    // EigenLayer contracts
+    DelegationFaucet delegationFaucet;
+
+    // M2 testing/mock contracts
+    ERC20PresetMinterPauser public stakeToken;
+    StrategyBase public stakeTokenStrat;
+
+    bytes32 public constant MINTER_ROLE = keccak256("MINTER_ROLE");
+    uint256 public constant DEFAULT_AMOUNT = 100e18;
+    address owner = cheats.addr(1000);
+
+    function setUp() public virtual override {
+        EigenLayerDeployer.setUp();
+
+        // Deploy ERC20 stakeToken, StrategyBase, and add StrategyBase to whitelist
+        stakeToken = new ERC20PresetMinterPauser("StakeToken", "STK");
+        stakeTokenStrat = StrategyBase(
+            address(
+                new TransparentUpgradeableProxy(
+                    address(baseStrategyImplementation),
+                    address(eigenLayerProxyAdmin),
+                    abi.encodeWithSelector(StrategyBase.initialize.selector, stakeToken, eigenLayerPauserReg)
+                )
+            )
+        );
+        cheats.startPrank(strategyManager.strategyWhitelister());
+        IStrategy[] memory _strategy = new IStrategy[](1);
+        _strategy[0] = stakeTokenStrat;
+        strategyManager.addStrategiesToDepositWhitelist(_strategy);
+        cheats.stopPrank();
+
+        // Deploy DelegationFaucet, grant it admin/mint/pauser roles, etc.
+        delegationFaucet = new DelegationFaucet(strategyManager, delegation, stakeToken, stakeTokenStrat);
+        stakeToken.grantRole(MINTER_ROLE, address(delegationFaucet));
+    }
+
+    /**
+     * @notice Assertions in test
+     * - Checks for staker contract deployed
+     * @param _operatorIndex is the index of the operator to use from the test-data/operators.json file
+     * @param _depositAmount is the amount of stakeToken to mint to the staker and deposit into the strategy
+     */
+    function test_mintDepositAndDelegate_DeployStakerContract(uint8 _operatorIndex, uint256 _depositAmount) public {
+        cheats.assume(_operatorIndex < 15 && _depositAmount < DEFAULT_AMOUNT);
+        if (_depositAmount == 0) {
+            _depositAmount = DEFAULT_AMOUNT;
+        }
+        // Setup Operator
+        address operator = getOperatorAddress(_operatorIndex);
+        address stakerContract = delegationFaucet.getStaker(operator);
+        _registerOperator(operator);
+
+        // Mint token to Staker, deposit minted amount into strategy, and delegate to operator
+        assertTrue(
+            !Address.isContract(stakerContract),
+            "test_mintDepositAndDelegate_DeployStakerContract: staker contract shouldn't be deployed"
+        );
+        delegationFaucet.mintDepositAndDelegate(operator, _depositAmount);
+        assertTrue(
+            Address.isContract(stakerContract),
+            "test_mintDepositAndDelegate_DeployStakerContract: staker contract not deployed"
+        );
```

Comment:
seems like here we can also check:
delegation status (staker now delegated to operator)
change in delegated amount (increase to operator, in strategy shares)
perhaps increase in deposited amount (shares that the staker has in strategy, seems like overkill though)

Reply:
ah, looks like some of these checks are in the `test_mintDepositAndDelegate_TokenBalances` test below?
personally I think if we're doing the exact same operations, it makes sense to combine these tests and just do all the checks in a single one.

Reply:
Updated with `test_mintDepositAndDelegate_CheckBalancesAndDeploys` test now